src/DBFTPlugin/ConsensusContext.cs

```diff
@@ -397,13 +411,16 @@ public ExtensiblePayload MakeRecoveryMessage()
             PrepareRequest prepareRequestMessage = null;
             if (TransactionHashes != null)
             {
+                var (proof, nonce) = GetNonce(keyPair.PrivateKey);
+                Block.Header.Nonce = nonce;
                 prepareRequestMessage = new PrepareRequest
                 {
                     Version = Block.Version,
                     PrevHash = Block.PrevHash,
                     ViewNumber = ViewNumber,
                     Timestamp = Block.Timestamp,
                     BlockIndex = Block.Index,
+                    VRFProof = proof, // Add vrf proof to the prepare request
```

Comment:
You should use the proof in `RrepareRequest` instead creating a new one. Proof should be caculated from Primary private key.

Reply:
Yes, creating a new one will cause issue when stopping the CNs at different stages. They can no longer achieve the consensus.
![image](https://user-images.githubusercontent.com/43694095/121654014-dd040000-cacf-11eb-893b-7d7d2491a4ba.png)

Reply:
> You should use the proof in `RrepareRequest` instead creating a new one. Proof should be caculated from Primary private key.

I did not get that, isn't that only the primary could send the recovery message?